### PR TITLE
React server

### PR DESCRIPTION
**Pull request by FarahZaqout**
create react server with create-react-app

* Eslint is disabled in app.jsx to keep it as a class that contains everything else (so we don't have to rewrite it later).

#26

### CODE REVIEW

**Comment by amusameh** [comment]
@FarahZaqout 
Why using .jsx while create-react-app uses .js?? 
>The distinction between .js and .jsx files was useful before Babel, but it’s not that useful anymore.
>Dan Abramov

you can read his comment is these issues
https://github.com/facebook/create-react-app/issues/87#issuecomment-234627904
https://github.com/airbnb/javascript/pull/985

File: client/public/index.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (RamyAlshurafa) [inline client/public/index.html]: I Think we should change this `title` into a real one

File: client/public/index.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <meta name="theme-color" content="#000000">
+    <!--
```
Comment (RamyAlshurafa) [inline client/public/index.html]: No needs for these commented lines.
what do you think?

File: client/src/index.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+import * as serviceWorker from './serviceWorker';
```
Comment (RamyAlshurafa) [inline client/src/index.jsx]: Do you want to make it off-line app?
if not so I think no needs for `serviceWorker`

Comment (FarahZaqout) [reply client/src/index.jsx]: I think it's good to try and make it a progressive app.

File: client/public/index.html (hunk 2 above)
Comment (FarahZaqout) [reply client/public/index.html]: Agreed.

File: client/public/index.html (hunk 1 above)
Comment (FarahZaqout) [reply client/public/index.html]: Yeah.

**Comment by FarahZaqout** [comment]
> @FarahZaqout
> Why using .jsx while create-react-app uses .js??
> 
> > The distinction between .js and .jsx files was useful before Babel, but it’s not that useful anymore.
> > Dan Abramov
> 
> you can read his comment is these issues
> [facebook/create-react-app#87 (comment)](https://github.com/facebook/create-react-app/issues/87#issuecomment-234627904)
> [airbnb/javascript#985](https://github.com/airbnb/javascript/pull/985)

Yeah, a lot of people are pushing towards this. However, for some reason airbnb refuses to buckle.
I have read the issues you linked before and the ones advocating for ```.js``` sound more reasonable. However, I would be lying to say that I know for sure what advantages and disadvantages that entails.

**Comment by FarahZaqout** [comment]
Maybe I should just disable the rule in eslint?

**Pull request merged**